### DISCUSSION TITLE
How to activate a template with autoexec.bat

### DISCUSSION THREAD

**Discussion by wagnerohz** [arendst/Tasmota#18696] [Q&A] [answered]
I have observed that on an ESP with several sensors (Higrow), the "activate" check box always empty when the battery runs empty. Now I want to use an autoexec.bat (or a better method) to do automatically on restart after charging, which I do: After the reboot I go into the console, check the box and all sensors are evaluated correctly.

**Comment by barbudor** [comment, marked as answer]
The command is 'module 0'
However you can't do that in autoexec.bat as this command it will restart the device... Indefinitively

You may do that with a rule that will first check if the module is already set to zero, and if not would set it.

The core problem you are facing is that when the battery is low, your device will start rebooting very quickly which will trigger the Fast Reboot Check.
You should probably solve this with a proper hardware design which prevent the device from Restarting when. The battery is below a proper voltage. If your battery is Liion, than would also reduce the risk of killing the battery by drowning it below it's security level.
A voltage supervisor chip would maintain the esp in reset while the battery voltage is below 2.90-3.00V

**Comment by wagnerohz** [comment]
Thank you very much for the really valuable suggestions and information. I had not even remotely thought of the Fast Reboot Check. I already measure the battery voltage, so now I have to learn how to build a rule that shuts down or stops the unit when the voltage is less than 3 volts.

**Reply by barbudor** [reply]
Indeed, you can solve that by software too if you can connect an ADC input to measure the voltage of the battery.
When detecting a too low battery level, just halt the ESP using `Restart 2` which will stop the ESP until an external reset.
